src/main/java/org/terasology/climateConditions/HyperthermiaSystem.java

```diff
@@ -73,4 +87,17 @@ public void onSpawn(OnPlayerSpawnedEvent event, EntityRef player, LocationCompon
         }
     }
 
+    private void weakenPlayer(EntityRef player, HealthComponent health) {
+        health.maxHealth *= healthReduceFactor;
+        health.currentHealth *= healthReduceFactor;
```

Comment:
With this implementation the current health will drop with the weakening effect kicking in - probably even beyond the new value of `maxHealth`. Is that intended?

For instance, if the player's health is at ♥️ 80/100 when the hyperthermia component is added it would result in the player's health being ♥️ 64/80. Imho it should be ♥️ 80/80, though.

Thus, I'd say the new `currentHealth` should be the minimum of `currentHealth` and the new `maxHealth`.

Reply:
I would argue that (1) is no good reason for the code being that way. This rather indicates that the curren UI system for health (and/or bars in general) is not flexible enough to express what we want to do here. 

Personally, I think that improving this could also become a part of your project, as the overall goal is to build a solid foundation of gameplay features, and I'd rather have a small but noticable improvement that many fancy features under the hood we cannot convey to the player.

I think (2) depends on how one subjectively looks at it. I'd argue that the bruises you mention are accounted for by having reduced health. Weakening the body now implies that I cannot recover from these bruises (get back to full health), but it does imply taking damage in some way. 

Thus, I'll stick to my suggestions as made yesterday 😉

Reply:
Yeah, well that does make more sense. Will look into the UI System then.
There's a tiny little thing I had in mind, just to keep in mind. As long as we do not have a temperature system which has levels of the extreme effects, this feature might become weird. Right now, the player shall weaken as soon as he/she enters the desert. Now, if the player is playing on the boundary of desert and forest, s/he shall essentially end up receiving a damage of 20 hit points if desert is entered and then exited from immediately after, since we won't be scaling back the health and let regen do the work.

Reply:
done.